src/core/tools/hpc/hpc_logger.cpp

```diff
@@ -0,0 +1,399 @@
+/*
+* The MIT License (MIT)
+*
+* Copyright (c) 2015 Microsoft Corporation
+*
+* -=- Robust Distributed System Nucleus (rDSN) -=-
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+*/
+
+
+/************************************************************
+*   hpc_logger (High-Performance Computing logger)
+*
+*	Editor: Chang Lou (v-chlou@microsoft.com)
+*
+*
+*	The structure of the logger is like the following graph.
+*
+*	For each thread:
+*	-------------------------              -------------------------
+*	|    new single log     |     ->       |                       |
+*	-------------------------              | --------------------- |
+*                                          |                       |
+*                                          | --------------------- |
+*                                          |        buffer         |
+*                                          |     (per thread)      |
+*                                          | --------------------- |
+*                                          |                       |
```

Comment:
the description here is not consistent with the implementation anymore, please fix or remove.